astronomer/airflow/version_check/update_checks.py

```diff
@@ -232,7 +284,8 @@ def parse_version(rel):
                 continue
             if ver <= current_version:
```

Comment:
If we want the current version (which we do now because we want to check for EOL etc), can't we just do switch this to `<` instead of having a whole new way to populate the current version?

I don't mean to be contradictory to my earlier comments - I missed the need for the current version in the db, but let's keep it simple (assuming that works and I'm not overlooking some other thing here).

Reply:
I think my previous implementation was the same. But then based on your comment I realised that we should separate checking updates and adding the current version in two different methods and call them separately in the run method. That was the reason to create a separate method for this.

Reply:
I've made the changes to the code as you suggested. However, this would also require us to update the logic for the available_update() method, right? Are you okay with that? What do you think? Personally, I think the approach I implemented before this comment is better.

Reply:
The reason the test fails is because now the current version is added in `check_for_update`, which that test calls, vs when you were adding it outside the check loop in `run`, which that test doesn't call. Either way, the issue the test highlights does exist though - the current version ends up in the db and not being hidden.

I see 2 options. First is to call `hide_old_versions` when we finish doing updates. Second would be to start setting the `hidden_from_ui` flag in `_process_update_json`. The second is probably better - no race conditions there.

Reply:
Yes second one makes better sense.